compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5629,23 +5635,22 @@ TR::Register* OMR::X86::TreeEvaluator::vectorCompareEvaluator(TR::Node* node, TR
          cg->stopUsingRegister(invMaskReg);
          }
 
-      node->setRegister(resultReg);
-
       if (maskReg)
          {
-         TR::Register *tmpReg = cg->allocateRegister(TR_VRF);
+         TR::InstOpCode andOpcode = TR::InstOpCode::PANDRegReg;
+         OMR::X86::Encoding andEncoding = andOpcode.getSIMDEncoding(&cg->comp()->target().cpu, vl);
+         TR_ASSERT_FATAL(andEncoding != OMR::X86::Bad, "Unsupported vector and opcode in vmcmp");
 
-         generateRegRegInstruction(movOpcode.getMnemonic(), node, tmpReg, resultReg, cg, movEncoding);
-         generateRegRegInstruction(movOpcode.getMnemonic(), node, resultReg, lhsReg, cg, movEncoding);
+         generateRegRegInstruction(andOpcode.getMnemonic(), node, resultReg, maskReg, cg, andEncoding);
```

Comment:
If I'm reading this correctly, I think `movOpcode` is only ever used [in the if block](https://github.com/eclipse-omr/omr/blob/209118f179f40cee3e31a1f1f59aadddc322cbc1/compiler/x/codegen/OMRTreeEvaluator.cpp#L5606) that appears above.  I would suggest moving its declaration there?

Reply:
Done